File: com/python/BasicPrograms/find_duplicates.py
arr = [1,2,3,4,5,1,1,3,4,6,8,9]

def find_duplicates(arr):
    res = {}
    for i in range(0, len(arr)):
        if arr[i] in res:
            res[arr[i]] = res[arr[i]] + 1
        else:
            res[arr[i]] = 1
    print res


def print_only_dups(arr):
    res = {}
    for i in range(0, len(arr)):
        if arr[i] in res:
            res[arr[i]] = res[arr[i]] + 1
        else:
            res[arr[i]] = 1
    for k,v in res.iteritems():
        if v > 1:
            print str(k) + " " + str(res[k])




print_only_dups(arr)

File: LeetCode/EasyJava/FindDuplicates.java
package Easy;

import java.util.ArrayList;
import java.util.HashMap;

public class FindDuplicates {

    public static ArrayList<Integer> getDups(int[] arr){
        HashMap<Integer, Integer> dups = new HashMap<>();
        if(arr.length == 0 || arr.length == 1)
            return null;
        for(int i=0; i < arr.length ; i++){
            if(dups.containsKey(arr[i]))
                dups.put(arr[i], dups.get(arr[i]) + 1);
            else
                dups.put(arr[i], 1);
        }
        ArrayList<Integer> res = new ArrayList<>();

        dups.forEach((k,v) -> {
            if(v > 1)
                res.add(k);
        });

        return res;
    }

    public static void main(String[] args) {
        int arr[] = {1,2,3,4,1,1,4,0,2,3};
        System.out.println(getDups(arr));
    }

}
